src/modules/cmdpal/CmdPalModuleInterface/dllmain.cpp

```diff
@@ -42,28 +43,24 @@ class CmdPal : public PowertoyModuleIface
     //contains the non localized key of the powertoy
     std::wstring app_key;
 
-    void LaunchApp()
+    void LaunchApp(const std::wstring& appPath, const std::wstring& commandLineArgs, bool elevated)
     {
-        auto package = package::GetRegisteredPackage(L"Microsoft.CommandPalette", false);
-
-        if (package.has_value())
+        std::wstring dir = std::filesystem::path(appPath).parent_path();
+
+        SHELLEXECUTEINFO sei = { 0 };
+        sei.cbSize = sizeof(SHELLEXECUTEINFO);
+        sei.hwnd = nullptr;
+        sei.fMask = SEE_MASK_NOCLOSEPROCESS | SEE_MASK_NO_CONSOLE;
+        sei.lpVerb = elevated ? L"runas" : L"open";
+        sei.lpFile = appPath.c_str();
+        sei.lpParameters = commandLineArgs.c_str();
+        sei.lpDirectory = dir.c_str();
+        sei.nShow = SW_SHOWNORMAL;
```

Comment:
if we instead stick `SW_HIDE` in here, and get rid of the arg, and instead get the `PInvoke.GetStartupInfo()`, can we see if SW_HIDE was set there, and not show the window in that case?

(I feel like I don't really want to add a magic hidden arg to our exe, but I'll accept it for now if it's the only way)

Reply:
I'll give it a try

Reply:
doesn't work. even if I pass SW_HIDE when launching, PInvoke says SW_SHOW is there. No idea why

Reply:
damn. Mmk. We can come up with something more clever in the future. Thanks for taking a look!